core/src/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -390,4 +393,29 @@ public static FileInputStream getFileInputStream( FileObject fileObject ) throws
     return new FileInputStream( fileObject.getName().getPathDecoded() );
   }
 
+  /**
+   * Returns a "friendly path", this is a path without a password.
+   */
+  public static String getFriendlyURI( String uri ) {
```

Comment:
Please make a new signature for getFriendlyURI( FileObject fileObject) that calls FileObject.getFriendlyURI().  Then, change this one to call this.getFriendlyURI(this.getFileObject(uri)) // minus the "this."

We want both signatures available to developers, but since they will already have a handle to a fileObject, we should cater to that.